data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/sink/KafkaSink.java

```diff
@@ -0,0 +1,166 @@
+
+    private final ExpressionEvaluator expressionEvaluator;
+
+    private final SinkContext sinkContext;
+
+
+    @DataPrepperPluginConstructor
+    public KafkaSink(final PluginSetting pluginSetting, final KafkaSinkConfig kafkaSinkConfig, final PluginFactory pluginFactory,
+                     final ExpressionEvaluator expressionEvaluator, final SinkContext sinkContext) {
+        super(pluginSetting);
+        this.pluginSetting = pluginSetting;
+        this.kafkaSinkConfig = kafkaSinkConfig;
+        this.pluginFactory = pluginFactory;
+        this.expressionEvaluator = expressionEvaluator;
+        this.sinkContext = sinkContext;
+
+
+    }
+
+
+    @Override
+    public void doInitialize() {
+        try {
+            doInitializeInternal();
+        } catch (InvalidPluginConfigurationException e) {
+            LOG.error("Invalid plugin configuration, Hence failed to initialize kafka-sink plugin.");
+            this.shutdown();
+            throw e;
+        } catch (Exception e) {
+            LOG.error("Failed to initialize kafka-sink plugin.");
+            this.shutdown();
+            throw e;
+        }
+    }
+
+    private void doInitializeInternal() {
+        executorService = Executors.newFixedThreadPool(totalWorkers);
+        sinkInitialized = Boolean.TRUE;
+    }
+
+    @Override
+    public void doOutput(Collection<Record<Event>> records) {
+        if (records.isEmpty()) {
+            return;
+        }
+        try {
+            final KafkaSinkProducer producer = createProducer();
+            records.forEach(record -> {
+                producerWorker = new ProducerWorker(producer, record);
+                executorService.submit(producerWorker);
+            });
+
+        } catch (Exception e) {
+            LOG.error("Failed to setup the Kafka sink Plugin.", e);
+            throw new RuntimeException(e.getMessage());
+        }
+    }
+
+    public KafkaSinkProducer createProducer() {
+        Properties properties = SinkPropertyConfigurer.getProducerProperties(kafkaSinkConfig);
+        properties = Objects.requireNonNull(properties);
+        return new KafkaSinkProducer(new KafkaProducer<>(properties),
+                kafkaSinkConfig, new DLQSink(pluginFactory, kafkaSinkConfig, pluginSetting), getSchemaRegistryClient(), expressionEvaluator
+                , Objects.nonNull(sinkContext) ? sinkContext.getTagsTargetKey() : null);
+    }
+
+    private CachedSchemaRegistryClient getSchemaRegistryClient() {
+        if (kafkaSinkConfig.getSchemaConfig() != null && kafkaSinkConfig.getSchemaConfig().getRegistryURL() != null) {
+            Properties schemaProps = new Properties();
+            SinkPropertyConfigurer.setSchemaProps(kafkaSinkConfig, schemaProps);
+            Map propertiesMap = schemaProps;
+            return new CachedSchemaRegistryClient(
+                    kafkaSinkConfig.getSchemaConfig().getRegistryURL(),
+                    100, propertiesMap);
```

Comment:
What is 100? can you please define a constant

Reply:
we will incorporate this change in next PR.